### PR TITLE
Make container/conda profiles more strict

### PR DESCRIPTION
**Pull request by jfy133**
This draft PR is to act as a PoC for making container/conda profiles more strict as to what is being inherited.

Originally described on [slack](https://nfcore.slack.com/archives/CGRTMASKY/p1613803009007800).

The purpose of the change is to ensure that, given nextflow inheritance rules, the final container/conda profile specified in `-profile` should be used. Currently `conda` is given a lower priority over containers, and if a container profile is specified somewhere in the profile list it will override the use of `conda` regardless if `conda` is specified last.

Use case: a user wants to run a pipeline using their institutional profile, where e.g. singularity is being used by default within the profile. However the user is test/developing a pipeline and would rather use conda.

## PR checklist

 - [x] This comment contains a description of changes (with reason)
 - [ ] `CHANGELOG.md` is updated
 - [ ] If you've fixed a bug or added code that should be tested, add tests!
 - [ ] Documentation in `docs` is updated

### CODE REVIEW

File: nf_core/pipeline-template/{{cookiecutter.name_noslash}}/nextflow.config [hunk 1]
```diff
@@ -57,21 +57,33 @@ try {
...
 
 profiles {
-  conda { process.conda = "$projectDir/environment.yml" }
+  conda { 
+    docker.enabled = false
+    singularity.enabled = false
+    podman.enabled = false
+    process.conda = "$projectDir/environment.yml" 
+    
+    }
```
Comment (ewels) [inline nf_core/pipeline-template/{{cookiecutter.name_noslash}}/nextflow.config:66]:
```suggestion
    process.conda = "$projectDir/environment.yml" 
  }
```

**Review by ewels — APPROVED** [review]
I can't think of / remember any reason _not_ to do this. So fine by me. 👍🏻

**Comment by MaxUlysse** [comment]
Was doing that in Sarek, works well, I think I removed it with the lastest sync

**Comment by jfy133** [comment]
Should I post in the slack channel with a @channel ping to make sur eit's not going to _break_ anything for anyone (maybe they have their own strategies, for whatever reason)? I feel this sort of a big change given _everyone_ uses these profiles...

**Comment by MaxUlysse** [comment]
or isn't it something that we would want to take care on within the configs? at an institutional level?

**Comment by jfy133** [comment]
I don't think so. The institutional example I gave is only one example, I think it could happen in other contexts with private configs, and the names of configs we are modifying here are most descriptive of the behaviour we are defining.

Unless I misunderstand?

**Comment by phue** [comment]
I think this needs a rebase because of additional profiles that were introduced in https://github.com/nf-core/tools/commit/bbd26abb1c21ed2b37a3491d96fce6c2ebff1c5e

File: nf_core/pipeline-template/{{cookiecutter.name_noslash}}/nextflow.config [hunk 2]
```diff
@@ -58,21 +58,32 @@ try {
...
     singularity.enabled = true
+    podman.enabled = false
     singularity.autoMounts = true
   }
   podman {
+    singularity.enabled = false
+    docker.enabled = false
     podman.enabled = true
   }
   shifter {
```
Comment (ewels) [inline nf_core/pipeline-template/{{cookiecutter.name_noslash}}/nextflow.config:89]: Needs `shifter` and `charliecloud` too..

**Comment by ewels** [comment]
> I think this needs a rebase because of additional profiles that were introduced in [bbd26ab](https://github.com/nf-core/tools/commit/bbd26abb1c21ed2b37a3491d96fce6c2ebff1c5e)

Nice spot @phue 👌🏻 I pulled in the changes for you @jfy133 and pushed to your branch. If you could add in the other container engines as well then I think we're good to go.

**Comment by jfy133** [comment]
Done! Just to double check @phue - charliecloud does requires it's own 'dedicated' config (where the `*.enabled` option is defined)?

**Comment by phue** [comment]
It *used* to be required to set the container environment explicitly using an `env` block. That's what the additional config file was for. I opened https://github.com/nf-core/tools/pull/867 to get rid of this.
So either you pull the commits in here or I'll fix that up once this is merged :+1:

**Review by ewels — APPROVED** [review]

**Pull request merged**